packages/itwin/grouping-mapping-widget/src/widget/hooks/useValidator.ts

```diff
@@ -15,13 +15,13 @@ export const Validators = {
   OnlyBeginsWithLetterOrUnderscore: {
     message: "Name can only start with a letter or underscore.",
     rule: (val: string) => {
-      return !val.match(/^[\W\d]+/i);
+      return !val.match(/^[\p{Nd}\p{S}\p{Z}\p{Po}]+/iu);
```

Comment:
The regexes do not look right. Here is the documentation which contains a regex for OData simple identifier:
http://docs.oasis-open.org/odata/odata/v4.0/errata03/os/complete/part3-csdl/odata-v4.0-errata03-os-part3-csdl-complete.html#_SimpleIdentifier

Mapping, Group, and GroupProperty names are OData simple identifiers.

The first character must match `[\p{L}\p{Nl}_]`
All the other characters must match `[\p{L}\p{Nl}\p{Nd}\p{Mn}\p{Mc}\p{Pc}\p{Cf}]`
Length should not exceed 128 characters.

Reply:
Thank you, Mindaugas, for pointing this out. I was struggling a bit with how String.prototype.match works. I just realized
that this function only returns an object of RegEx instead of a boolean value (which is required for the IRule interface when initializing the **validator** variable). As first I tried something like this:
![image](https://github.com/iTwin/viewer-components-react/assets/56598021/8e1061cc-70a7-4398-8d10-2388b4f15fd4)

But then the UI would return true by default since the validator returns an object. The backend instead did not take this input and returned a null name, which caused an error of "Error! Something went wrong!", but the UI was supposed to return the "Name can only start with a letter or an underscore." Therefore, I tried the approach of !val.match(...) so that the validator could output a boolean variable.

However, the Regex.prototype.test() actually returns a boolean output so I did figure out how to approach the problem properly by not sticking to match() in my second commit.